scripts/featured

```diff
@@ -319,7 +319,20 @@ class FeatureHandler(object):
         Returns:
             None.
         """
-        restart_field_str = "always" if "enabled" in feature_config.auto_restart else "no"
+        # As per the current code(due to various dependencies) SWSS service stop/start also stops/starts the dependent services(syncd, teamd, bgpd etc)
+        # There is an issue seen of syncd service getting stopped twice upon a critical process crash in syncd service due to above reason.
+        # Also early start of syncd service has traffic impact on VOQ chassis.
+        # to fix the above issue, we are disabling the auto restart of syncd service as it will be started by swss service.
+        # This change can be extended to other dependent services as well in future and also on pizza box platforms.
+
+        device_type = self._device_config.get('DEVICE_METADATA', {}).get('localhost', {}).get('type')
+        is_dependent_service = feature_config.name in ['syncd', 'gbsyncd']
+        if device_type == 'SpineRouter' and is_dependent_service:
+            syslog.syslog(syslog.LOG_INFO, "Skipped setting Restart field in systemd for {}".format(feature_config.name))
```

Comment:
question: This change similar to making autorestart configuration disabled for `syncd` feature for Spine devices, is that correct?.  Can we update the init_config.j2 file with condition iso adding condition here? will that work?

Reply:
>question: This change similar to making autorestart configuration disabled for syncd feature for Spine devices, is that correct?. Can we update the init_config.j2 file with condition iso adding condition here? will that work?

Not entirely.. We are not disabling auto restart feature completely here.. There are other side effects to that.. for eg: if we disable auto restart for syncd, any critical process crash wont exit the service and the service wont get restarted again(there is autorestart feature check in supervisor-exit-process-listener)
Code snippet:
if (process_name in critical_process_list or group_name in critical_group_list) and expected == 0:
                    is_auto_restart = get_autorestart_state(container_name)
                    if is_auto_restart != "disabled":
                        MSG_FORMAT_STR = "Process '{}' exited unexpectedly. Terminating supervisor '{}'"
                        msg = MSG_FORMAT_STR.format(payload_headers['processname'], container_name)
                        syslog.syslog(syslog.LOG_INFO, msg)
                        publish_events(events_handle, payload_headers['processname'], container_name)
                        swsscommon.events_deinit_publisher(events_handle)
                        os.kill(os.getppid(), signal.SIGTERM)
                    else:
                        process_under_alerting[process_name]["last_alerted"] = time.time()
                        process_under_alerting[process_name]["dead_minutes"] = 0